### PR TITLE
 add functionality to filter buttons

### PR DESCRIPTION
**Pull request by wellingtonmapise**
# Description

This PR adds two key UI features for post filtering:

* **Recent button** – enables sorting posts by latest
* **Type filter button** – toggles between `OFFER` and `REQUEST` posts

These improve the browsing experience by helping users quickly find relevant or recent posts.
---

# Milestones

* Browse and filter posts by type
* Improve post discoverability via recency
* Lays groundwork for combined filter/search UX

---

# Resources

* Figma reference: https://www.figma.com/design/4i5kmsSfGnfDx4byG9CKHG/SkillSwap
* All filter interactions built manually (no 3rd-party libs)

---

# Test Plan

* Manually verified button toggles update UI as expected
* Confirmed post list dynamically updates on each filter change
* Verified behavior when no posts match selected filter

**Edge cases tested:**
* Switching filters when no posts exist for a type
* Posts missing a `type` field
* Repeated toggling between filters

### CODE REVIEW

File: frontend/src/pages/LandingPage.jsx [hunk 1]
```diff
@@ -38,6 +40,25 @@ const LandingPage = () => {
     fetchRecommendedPosts();
   }, []);
 
+  const filteredPosts = posts.filter((post) => {
```
Comment (juliannelin213) [inline frontend/src/pages/LandingPage.jsx:43]: nit: maybe you should move this to utils?

File: frontend/src/pages/LandingPage.jsx [hunk 2]
```diff
@@ -38,6 +40,25 @@ const LandingPage = () => {
...
+    const matchType = type ? post.type === type : true;
+    const matchRecency = recency
+      ? (() => {
+          const createdAt = new Date(post.createdAt);
+          const now = new Date();
+          if (recency === 'last_week') {
```
Comment (juliannelin213) [inline frontend/src/pages/LandingPage.jsx:49]: nit: I think you can use switch cases for readability

File: frontend/src/components/Filters/FilterBar.jsx [hunk 3]
```diff
@@ -9,15 +9,21 @@ function FilterBar({ setFilter }) {
...
       </button>
-      <select className="filter-select">
+      <select
+        className="filter-select"
+        onChange={(e) => setType(e.target.value)}
+      >
         <option value="">Type</option>
-        <option value="Offer">Offer</option>
-        <option value="Request">Request</option>
+        <option value="OFFER">Offer</option>
```
Comment (juliannelin213) [inline frontend/src/components/Filters/FilterBar.jsx:17]: nit: why is this capitalized here and other values are lower cases, maybe we can keep it consistent
